### PR TITLE
Support Rails 7 on attr_encrypted

### PR DESCRIPTION
**Pull request by E70M**
### Pivotal story
https://www.pivotaltracker.com/story/show/184644060

### Contributing guide checklist:
- [X] The first commit follows the contributing guide
- [X] PR title follows the contributing guide
- [X] Code has appropriate test coverage
- [X] I have performed a self-review of this code
- [X] I have added a link in the Pivotal story to this PR
- [X] I have requested 3 reviewers
- [X] I have added an assignee

### Context
- https://github.com/attr-encrypted/attr_encrypted/pull/425

This gem has an attribute `encrypted_attributes` whose name collides with an attribute in native [Rails ActiveRecord Encryption](https://guides.rubyonrails.org/active_record_encryption.html), and I believe some of the provided getters/setters do as well. The PR linked was opened on the upstream and does seem to work for 6.x and 7.x, but not 5.x, which is why it hasn't been merged yet (well, that and the project hasn't had a stable maintainer in a while). These changes appear to work against our main app on 6.x, and since we don't need 5.x support, I think we can move forward with the Rails 7 upgrade. If any other issues come up from this gem when upgrading to 7.x, we can follow up in that story

### Note for merger
I also fast-forwarded this branch to have the latest from upstream, so when merging, please use the rebase option rather than squash. I'll have to keep squashing code review changes into one commit to account for that

### CODE REVIEW

**Review by kceb — APPROVED** [review]

File: test/legacy_attr_encrypted_test.rb [hunk 1]
```diff
@@ -201,7 +201,7 @@ def test_should_use_options_found_in_the_attr_encrypted_options_attribute
   end
 
   def test_should_inherit_encrypted_attributes
-    assert_equal [LegacyUser.encrypted_attributes.keys, :testing].flatten.collect { |key| key.to_s }.sort, LegacyAdmin.encrypted_attributes.keys.collect { |key| key.to_s }.sort
+    assert_equal [LegacyUser.encrypted_attributes.keys, :testing].flatten.collect { |key| key.to_s }.sort, LegacyAdmin.attr_encrypted_encrypted_attributes.keys.collect { |key| key.to_s }.sort
```
Comment (arettberg) [inline test/legacy_attr_encrypted_test.rb:204]: missed one - `attr_encrypted_encrypted_attributes`

**Review by arettberg — APPROVED** [review]

File: test/test_helper.rb [hunk 2]
```diff
@@ -1,23 +1,20 @@
 # frozen_string_literal: true
 
+require 'pry'
```
Comment (kceb) [inline test/test_helper.rb:3]: https://github.com/simplepractice/attr_encrypted/pull/1/files#r1140461645

File: attr_encrypted.gemspec [hunk 3]
```diff
@@ -42,20 +39,16 @@ Gem::Specification.new do |s|
...
-    s.add_development_dependency('public_suffix', '< 3.0.0')
-  end
   if defined?(RUBY_ENGINE) && RUBY_ENGINE.to_sym == :jruby
     s.add_development_dependency('activerecord-jdbcsqlite3-adapter')
     s.add_development_dependency('jdbc-sqlite3', '< 3.8.7') # 3.8.7 is nice and broke
   else
     s.add_development_dependency('sqlite3')
   end
   s.add_development_dependency('dm-sqlite-adapter')
+  s.add_development_dependency('pry')
```
[the remark this answers is not in the document]
Comment (E70M) [reply attr_encrypted.gemspec:49]: It appears to be on the upstream (I didn't add it): https://github.com/simplepractice/attr_encrypted/pull/1/files/f3355d4cf4e2dbaf9a8ec97edffcc58e652f2e68

**Comment by kceb** [comment]
Seems like this is ready to merge

**Review by pdoulatshahi — APPROVED** [review]

**Pull request merged**